lib/Server.js

```diff
@@ -674,20 +674,15 @@ class Server {
   }
 
   createSocketServer() {
-    const socket = sockjs.createServer({
-      // Use provided up-to-date sockjs-client
-      sockjs_url: '/__webpack_dev_server__/sockjs.bundle.js',
-      // Limit useless logs
-      log: (severity, line) => {
-        if (severity === 'error') {
-          this.log.error(line);
-        } else {
-          this.log.debug(line);
-        }
-      },
+    const socket = new SockjsServer({
+      error: this.log.error.bind(this),
+      debug: this.log.debug.bind(this),
+      server: this.listeningApp,
+      path: this.sockPath,
```

Comment:
Better always pass all options (this.options)

Reply:
@evilebottnawi Are you suggesting, in addition to what is already there, `options: this.options`? Because `this.listeningApp` is not part of the options, along with the `error`, `debug` callbacks.

Users may be creating server implementations of their own, so they would be interacting with the options passed in. Do you think it's a good idea to expose all the options to the user?

Reply:
@Loonride 
> Do you think it's a good idea to expose all the options to the user?

Yes, we because behavior can be changed based on options (not only) in theory. 

Let's do it like:
`new SockJSServer(this);`

Custom implementation should have access to compiler, application, server, options and devServer.

Also it is not available for developers right now, so we can change this in future.